ocaml/idl/datamodel_vtpm.ml

```diff
@@ -53,6 +53,28 @@ let set_contents =
       ]
     ~hide_from_docs:true ~allowed_roles:_R_LOCAL_ROOT_ONLY ()
 
+let operations =
+  Enum
+    ( "vtpm_operations"
+    , [
+        ("destroy", "Destroy a VTPM")
+      ; ("get_contents", "Read data from a VTPM")
+      ; ("set_contents", "Write to the VTPM, overwriting existing content")
+      ]
+    )
+
+let assert_operation_valid =
```

Comment:
Do we need this function in the API? The GUI normally uses the `allowed_operations` fields in preference to function calls, and we don't have similar functions in the API for other devices (though VM does have it, for some reason).

Reply:
Updates to those fields as well as API-level locking is managed through message_forwarding. See e.g. how `with_vif_marked` is used in message_forwarding.ml.

Reply:
Similarly, you should use `with_vm_operation` in `message_forwarding.ml` on the `VTPM.create` function to ensure that the VM operations fields are set correctly, and the check is enforced on the create function.

Reply:
This is now unused (not mentioned in `~messages` below). I think we can remove it, because XC is unlikely to use it.